### DISCUSSION TITLE
Dependabot cannot update this dependency (Private Registry)

### DISCUSSION THREAD

**Discussion by zachwright** [community/community#23371] [New to GitHub]
<p>Using the .dependabot/config.yml file for configuring Dependabot on our projects that use private registries such as <a href="http://php.fury.io" rel="nofollow noopener">php.fury.io</a> works fine, but when using the .github/dependabot.yml file instead, GitHub says:</p>
<p>“Dependabot doesn’t support updating dependency files that use private package registries.”</p>
<p>Do we know if there will be support for this in the future? Is there action I can take now to make it work in the mean time?</p>

**Comment by AndreaGriffiths11** [comment, marked as answer]
<p>Hi <a class="mention" href="/zachwright">@zachwright</a>,<br>
Welcome to the community! We’re working adding private registry support over the coming months. For now, I would recommend you keep using Dependabot Preview. You can migrate back by removing the new config file and adding the repository in <a href="https://app.dependabot.com/">https://app.dependabot.com/</a></p>

**Comment by jef** [comment]
<p>Also waiting for this. Is this any time frame?</p>

**Comment by vinayakkulkarni** [comment]
<p>Would really like an update… I migrated from <code>dependabot-preview</code> to <code>dependabot</code> but now it won’t update the packages</p>

**Comment by agaudreault** [comment]
<aside class="onebox githubissue">
  <header class="source">
      <a href="https://github.com/github/roadmap/issues/67" target="_blank" rel="nofollow noopener">github.com/github/roadmap</a>
  </header>
  <article class="onebox-body">
    <div class="github-row">
  <div class="github-icon-container" title="Issue">
	  <svg width="60" height="60" class="github-icon" viewbox="0 0 14 16" aria-hidden="true"><path d="M7 2.3c3.14 0 5.7 2.56 5.7 5.7s-2.56 5.7-5.7 5.7A5.71 5.71 0 0 1 1.3 8c0-3.14 2.56-5.7 5.7-5.7zM7 1C3.14 1 0 4.14 0 8s3.14 7 7 7 7-3.14 7-7-3.14-7-7-7zm1 3H6v5h2V4zm0 6H6v2h2v-2z"></path></svg>
  </div>

  <div class="github-info-container">
    <h4>
      <a href="https://github.com/github/roadmap/issues/67" target="_blank" rel="nofollow noopener">Dependabot support for private registries</a>
    </h4>

    <div class="github-info">
      <div class="date">
        opened <span class="discourse-local-date" data-format="ll" data-date="2020-07-24" data-time="17:15:14" data-timezone="UTC">05:15PM - 24 Jul 20 UTC</span>
      </div>


      <div class="user">
        <a href="https://github.com/github-product-roadmap" target="_blank" rel="nofollow noopener">
          <img alt="github-product-roadmap" src="https://user-images.githubusercontent.com/47184027/181076590-f8e44a2e-0faf-4397-bbe0-728443c237b0.png" class="onebox-avatar-inline" width="20" height="20">
          github-product-roadmap
        </a>
      </div>
    </div>
  </div>
</div>

<div class="github-row">
  <p class="github-content">Summary
Dependabot support for private registries and private git repos allows users to keep private repos and packages secure by opening pull...</p>
</div>

<div class="labels">
    <span style="display:inline-block;margin-top:2px;background-color: #B8B8B8;padding: 2px;border-radius: 4px;color: #fff;margin-left: 3px;">all</span>
    <span style="display:inline-block;margin-top:2px;background-color: #B8B8B8;padding: 2px;border-radius: 4px;color: #fff;margin-left: 3px;">beta</span>
    <span style="display:inline-block;margin-top:2px;background-color: #B8B8B8;padding: 2px;border-radius: 4px;color: #fff;margin-left: 3px;">cloud</span>
    <span style="display:inline-block;margin-top:2px;background-color: #B8B8B8;padding: 2px;border-radius: 4px;color: #fff;margin-left: 3px;">security &amp; compliance</span>
</div>

  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

**Comment by agos** [comment]
<p>GitHub announced yesterday that Dependabot <a href="https://github.blog/changelog/2020-12-02-dependabot-version-updates-from-private-github-repositories/" rel="noopener nofollow ugc">now supports private repositories</a>. I tried this today with an npm package associated with a private repository with no luck. I suspect it works with private repositories (as in: a dependency which is a GitHub URL for a private repo) and not for packages in private repositories. I’ve filed a feedback hoping for some clarification, hopefully in the form of some update in the documentation.</p>

**Comment by David-Guillot** [comment]
<p>Hi there,</p>
<p>Despite <a href="https://github.blog/changelog/2020-12-02-dependabot-version-updates-from-private-github-repositories/" rel="noopener nofollow ugc">this blog post</a>, <a href="https://docs.github.com/en/free-pro-team@latest/github/administering-a-repository/about-dependabot-version-updates#supported-repositories-and-ecosystems" rel="noopener nofollow ugc">the documentation</a> still shows both private registries and private repositories not being supported. Plus, <a href="https://github.com/github/roadmap/issues/67" rel="noopener nofollow ugc">the public roadmap issue for this feature</a> was closed with label <code>shipped</code> for only 4 days: it’s now reopened, the <code>shipped</code> label has been removed, but it’s still in the “Q4 2020” column of <a href="https://github.com/github/roadmap/projects/1#card-42486614" rel="noopener nofollow ugc">the public roadmap</a>. <strong>Plus</strong>, and that’s the biggest one: old Dependabot app seems to have stopped supporting private repositories (at least in my case, for a Python/pip project requiring private Github repositories), which invalidates <a class="mention" href="/AndreaGriffiths11">@AndreaGriffiths11</a>’s answer.</p>
<p>Can we have some piece of information about:</p>
<ul>
<li>The scope of this feature: Private registries? Private repositories? Both?</li>
<li>What happened? Was the feature deployed then rolled back?</li>
<li>What’s going to happen now on that matter?</li>
</ul>

**Comment by BorntraegerMarc** [comment]
<p>However it seems like GitHub is distinguishing between private external registries and private GitHub repos: <a href="https://github.com/github/roadmap/issues/155" class="inline-onebox" rel="noopener nofollow ugc">Dependabot version updates: support for private repositories (Cloud Beta) · Issue #155 · github/roadmap · GitHub</a></p>

**Comment by BorntraegerMarc** [comment]
<p>About that: Using dependabot with the private github registry isn’t generally working for me. However editing the content of the <code>.npmrc</code> file from:</p>
<pre><code class="lang-auto">registry=https://npm.pkg.github.com/komed-health
</code></pre>
<p>To:</p>
<pre><code class="lang-auto">@komed-health:registry=https://npm.pkg.github.com
</code></pre>
<p>As described here: <a href="https://docs.github.com/en/packages/guides/configuring-npm-for-use-with-github-packages#installing-packages-from-other-organizations" class="inline-onebox" rel="noopener nofollow ugc">Configuring npm for use with GitHub Packages - GitHub Docs</a></p>
<p>Will at least let dependabot check for the other packages not in the private registry</p>

**Comment by mikefrancis** [comment]
<p>Bumping <a class="mention" href="/david-guillot">@david-guillot</a>’s comment - this does not work yet, which is odd given the blog post is still up.</p>

**Comment by David-Guillot** [comment]
<p>Some news:</p>
<ul>
<li>The <a href="https://github.com/github/roadmap/issues/67" rel="noopener nofollow ugc">original issue</a> has been marked as <code>shipped</code> again today</li>
<li>A <a href="https://github.blog/changelog/2021-03-15-dependabot-private-registry-support-public-beta/" rel="noopener nofollow ugc">new blog post</a> has been published</li>
</ul>
<p>This new feature seems to be related to private registries, which seems to be supported for all languages.</p>
<p>Doesn’t fit my needs (private repositories for Python/pip), but could be useful for some :slight_smile:</p>

**Comment by BorntraegerMarc** [comment]
<p>The feature is shipped but now I have this issue: <a href="https://github.community/t/dependabot-doesnt-see-github-actions-secrets/167104/32" class="inline-onebox">Dependabot doesn't see GitHub actions secrets - #32 by ryanhiebert</a></p>

**Comment by rribeiro1** [comment]
<p>Same here, it looks like the dependabot secrets are accessible only for dependabot configuration in <code>dependabot.yml</code>. For actions, it is not working :pensive:</p>

**Comment by mroach** [comment]
<p>I’ve had to read the <a href="https://github.blog/2021-03-15-dependabot-private-dependencies/" rel="noopener nofollow ugc">blog post</a>, <a href="https://docs.github.com/en/organizations/keeping-your-organization-secure/managing-security-and-analysis-settings-for-your-organization#allowing-dependabot-to-access-private-dependencies" rel="noopener nofollow ugc">docs</a>, and <a href="https://docs.github.com/en/code-security/supply-chain-security/keeping-your-dependencies-updated-automatically/configuration-options-for-dependency-updates#configuration-options-for-private-registries" rel="noopener nofollow ugc">dependabot.yml docs</a> several times because it seems that, and I can’t quite believe it, you once again need to generate <em>Personal</em> Access Tokens to access  private repositories and packages. that are all part of the same GitHub organisation?</p>
